src/commands/pressable-grant-access.php

```diff
@@ -2,91 +2,362 @@
 		$this->api_helper = new API_Helper();
 		$this->output     = $output;
 
+		// Handle if searching by domain or using the passed site name.
+		if ( $input->getOption( 'search' ) ) {
+			$this->handle_search_for_site( $input, $output );
+		} else {
+			$this->handle_site_by_name( $input, $output );
+		}
+
+		$output->writeln( "<info>\nAll done!<info>" );
+	}
+
+	/***********************************
+	 ***********************************
+	 *          INPUT GETTERS          *
+	 ***********************************
+	 ***********************************/
+
+	/**
+	 * Gets the email address either form Input or prompted for.
+	 *
+	 * @param \Symfony\Component\Console\Input\InputInterface $input
+	 * @param \Symfony\Component\Console\Output\OutputInterface $output
+	 * @return string
+	 */
+	private function get_email( InputInterface $input, OutputInterface $output ): string {
+		// Attempt to get the email from the Input.
 		$email = $input->getOption( 'email' );
-		$site = $input->getOption( 'site' );
 
+		// If we don't have an email, prompt the user.
 		if ( empty( $email ) ) {
-			$email = trim( readline( 'Please enter the email: ' ) );
-			if ( empty( $email ) ) {
-				$output->writeln( '<error>Missing email (--email=someone@a8c.com).</error>' );
-				exit;
-			}
+			$email = $this->get_question_helper()->ask( $input, $output, $this->ask_for_email_address() );
 		}
 
-		if ( empty( $site ) ) {
-			$site = trim( readline( 'Please enter the site: ' ) );
-			if ( empty( $site ) ) {
-				$output->writeln( '<error>Missing site (--site=my-webiste.com).</error>' );
-				exit;
-			}
+		// If we still don't have en email fail.
+		if ( empty( $email ) ) {
+			$output->writeln( '<error>You must supply a valid email address.</error>' );
+			exit;
 		}
 
-		$output->writeln( '<comment>Granting ' . $email . ' access to site ' . $site . '.</comment>' );
+		return $email;
+	}
 
-		$site_id = null;
-		if ( is_numeric( $site ) ) {
-			$site_id = $site;
-		} else {
-			$search_site = $this->api_helper->call_pressable_api( sprintf('sites', $site), 'GET', array() );
-			$output->writeln( '<comment>Looping through ' . count($search_site->data) . ' domains searching for ' . $site . '.</comment>' );
-
-			// Loop to find site_id by domain
-			foreach ( $search_site->data as $key => $val ) {
-				if ( $val->url === $site) {
-					$site_id = $val->id;
-					$output->writeln( "<comment>Found it! It's site ID {$site_id}</comment>" );
-					break;
-				}
+	/**
+	 * Gets the search term either form Input or prompted for.
+	 *
+	 * @param \Symfony\Component\Console\Input\InputInterface $input
+	 * @param \Symfony\Component\Console\Output\OutputInterface $output
+	 * @return string
+	 */
+	private function get_search_term( InputInterface $input, OutputInterface $output ): string {
+		// Attempt to get search term from Input.
+		$search_term = $input->getOption( 'search' );
+
+		// If we don't have a search term, prompt the user.
+		if ( empty( $search_term ) ) {
```

Comment:
Is there a scenario where there's an empty `$search_term` here? The function `get_search_term` is invoked by `handle_search_for_site`, which is only executed if `$input->getOption( 'search' )` has a value.

Reply:
Yes **this has now been removed,** it's left over from some earlier iteration, which allow --search but no value, but that became a dead path after other changes.